### ISSUE TITLE
Error during destruction of the Editing-instance

### ISSUE THREAD

**Issue by Snuffleupagus** [mozilla/pdf.js#15149]
Attach (recommended) or Link to PDF file here: N/A

Configuration:
- Web browser and its version: N/A
- Operating system and its version: N/A
- PDF.js version: Any version after PR #15130
- Is a browser extension: N/A

Steps to reproduce the problem:
1. Open the development viewer locally, after running `gulp server`, http://localhost:8888/web/viewer.html
2. Enable FreeText editing. (Note that Ink editing is similarly affected.)
3. Click on any page, and add a new comment (e.g. "Hello world").
4. Open the web-console, and run `PDFViewerApplication.close()`.

What is the expected behavior? (add screenshot)
Closing works correctly.

What went wrong? (add screenshot)
An error is thrown:
```
Uncaught (in promise) TypeError: editor.destroy is not a function
    destroy http://localhost:8888/src/display/editor/tools.js:439
    setDocument http://localhost:8888/web/base_viewer.js:644
    close http://localhost:8888/web/app.js:866
    async* debugger eval code:1
```

Basically, the issue is that none of the `AnnotationEditor`-classes implement a `destroy`-method. The simplest solution would be to *just remove* this code: https://github.com/mozilla/pdf.js/blob/b0a3c9e8cf84e3d20e754c6c2ab6650f3928d62d/src/display/editor/tools.js#L438-L440

Another solution would be to actually implement destroy-functionality for the `AnnotationEditor`-classes, but I'm not sure if that's actually necessary here (given that we're tearing down the current Editing-instance)?

/cc @calixteman What solution is "correct" here? If it's the first one I could definitely submit a PR to fix it.

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension): N/A
